### PR TITLE
feat(api): MultiQueryRetriever with WeaviateHybridSearchRetriever

### PR DESCRIPTION
**Pull request by Lee-W**
Contradict to our earlier discussion; I suspect what we want to replace `WeaviateHybridSearchRetriever` with is `MultiQueryRetriever` instead of `ConversationalRetrievalChain`. However, it seems to perform even worse than the `MultiQueryRetriever` when it comes to **what is astro python sdk?**. astronomer-providers, airflow things seem to work. For testing, we can use the same set of env var and the exact same way to bring up the test env locally.

### CODE REVIEW

**Comment by sunank200** [comment]
@Lee-W Now that we have `MultiQueryRetriever` with `WeaviateHydridSearchRetriever`. We should try the following two at the earliest when you start the day as Rahul needs to run the test for these.

We should try the following:
- Increase the number of documents we retrieved per rewording to 5
- when joining the docs together, we should rerank using something like [cohere reranker](https://python.langchain.com/docs/integrations/retrievers/cohere-reranker)

Let me know if you need any you need to connect on this, please

File: api/ask_astro/chains/answer_question.py [hunk 1]
```diff
@@ -12,9 +12,10 @@
     SystemMessagePromptTemplate,
 )
 from langchain.retrievers import MultiQueryRetriever
+from langchain.retrievers.weaviate_hybrid_search import WeaviateHybridSearchRetriever
 
-from ask_astro.clients.weaviate_ import docsearch
```
Comment (jlaneve) [inline api/ask_astro/chains/answer_question.py:16]: is this now dead code?

https://github.com/astronomer/ask-astro/blob/hybrid-search/api/ask_astro/clients/weaviate_.py#L29-L35

Comment (sunank200) [reply api/ask_astro/chains/answer_question.py:16]: Yes

**Comment by Lee-W** [comment]
> Increase the number of documents we retrieved per rewording to 5

tested with 6. a bit better but might have room to improve

> when joining the docs together, we should rerank using something like [cohere reranker](https://python.langchain.com/docs/integrations/retrievers/cohere-reranker)

not yet have time to work on it today. will continue work on it early tomorrow

**Comment by Lee-W** [comment]
close this one, as changes included in https://github.com/astronomer/ask-astro/pull/170

**Pull request closed without merging**